cmake/add_library_macros.cmake

```diff
@@ -10,11 +10,103 @@
 include(CMakeParseArguments)
 include(GenerateExportHeader)
 include(CheckLibArchitecture)
+include(CheckCXXCompilerFlag)
 
+#------------------------------------------------------------------------------
+# Function to add OpenMS compiler flags to any target
+#------------------------------------------------------------------------------
+function(openms_add_compiler_flags TARGET_NAME)
+  if (CMAKE_COMPILER_IS_GNUCXX)
+    target_compile_options(${TARGET_NAME} PRIVATE
+      -Wall
+      -Wextra
+      -Wno-non-virtual-dtor
+      -Wno-unknown-pragmas
+      -Wno-long-long
+      -Wno-unknown-pragmas
+      -Wno-unused-function
+      -Wno-variadic-macros)
+
+    if (ENABLE_GCC_WERROR)
+      target_compile_options(${TARGET_NAME} PRIVATE -Werror)
+    endif()
+
+    if (CMAKE_GENERATOR STREQUAL "Eclipse CDT4 - Unix Makefiles")
+      target_compile_options(${TARGET_NAME} PRIVATE -fmessage-length=0)
+    endif()
+  endif()
+
+  if (MSVC)
+    target_compile_options(${TARGET_NAME} PRIVATE
+      /wd4251 /wd4275  # disable dll-interface warning
+      /wd4996  # disable deprecated functions warning
+      /wd4661  # disable explicit template instantiation warning
+      /wd4503  # disable decorated name length warning
+      /wd4068  # disable unknown pragma warning
+      /bigobj  # allow large object files
+      /MP)     # use multiple CPU cores
+    
+    target_compile_definitions(${TARGET_NAME} PRIVATE
+      _SCL_SECURE_NO_WARNINGS
+      _CRT_SECURE_NO_WARNINGS
+      _CRT_SECURE_NO_DEPRECATE
+      OPENMS_XERCESDLL
```

Comment:
Some compile definitions you might want to have PUBLIC to propagate to downstream targets that are e.g. not libraries added by this macro (e.g. topp tools).

Reply:
- so better move function(openms_add_compiler_flags TARGET_NAME) into a separate cmake file?
- good point. will try to propagate
- what I did not fully understand: do I need the project level flags at all for any third-party etc.?